src/fund_migration.rs

```diff
@@ -19,25 +24,127 @@ pub enum MigrationStatus {
 }
 
 #[allow(dead_code)]
-fn sign_with_ldk_key(seed: [u8; 64], message: &str) -> Result<(String, String)> {
+pub(crate) async fn migrate_funds(
+    seed: &[u8; 64],
+    data_store: Arc<Mutex<DataStore>>,
+    sdk: &BreezServices,
+) -> Result<()> {
+    if matches!(
+        data_store
+            .lock()
+            .unwrap()
+            .retrive_funds_migration_status()?,
+        MigrationStatus::Completed | MigrationStatus::NotNeeded
+    ) {
+        return Ok(());
+    }
+
+    let (private_key, public_key) = derive_ldk_keys(seed)?;
+    let public_key = public_key.serialize().to_hex();
+
+    let balance = fetch_balance(public_key.clone())?;
+    if balance == 0 {
+        data_store
+            .lock()
+            .unwrap()
+            .append_funds_migration_status(MigrationStatus::NotNeeded)?;
+        return Ok(());
+    }
```

Comment:
From my understanding of the migration service, it will return errors (distinct ones) if the node never had any funds to migrate or if it already migrated. This snippet assumes it would return 0, at least in the former case.